src/NumericalAlgorithms/Spectral/ComplexDataView.hpp

```diff
@@ -0,0 +1,174 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <complex>
+#include <cstddef>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Spectral {
+/// Namespace for spin-weighted spherical harmonic utilities.
+namespace SWSH {
+
+/// \brief A set of labels for the possible representations of complex numbers
+/// for the `ComplexDataView` and the computations performed in the
+/// spin-weighted spherical harmonic transform library.
+///
+/// \details The representation describes one of two behaviors:
+///  - `Interleaved`: The vectors of complex numbers will primarily be
+///  represented by alternating doubles in memory. This causes both the real and
+///  imaginary part at a given gridpoint to be near one another, but successive
+///  real values farther. This is the native representation of complex data in
+///  the c++ standard, and is the representation needed for Blaze math
+///  operations. Therefore, using this representation type in libsharp
+///  computations will cause operations which access only the real or imaginary
+///  parts individually to trace over larger memory regions. However, this
+///  representation will require fewer copies.
+///  - `RealsThenImags`: The vectors of complex numbers will primarily be
```

Comment:
I'll add the note about libsharp. It can't be always because it needs to communicate data back to the  vectors of `std::complex<double>` representations in order to do vector math, because that's the representation Blaze speaks in.